macros/core/PGML.pl

```diff
@@ -1523,17 +1501,15 @@ sub Escape {
 }
 
 sub Indent {
-	my $self = shift;
-	my $item = shift;
+	my ($self, $item, $block, $state) = @_;
 	return $self->string($item) if $item->{indent} == 0;
-	my $em = 2.25 * $item->{indent};
-	return
-		$self->nl
-		. '<div style="margin:0 0 0 '
-		. $em . 'em">' . "\n"
-		. $self->string($item)
-		. $self->nl
-		. "</div>\n";
+	my $em     = 2.25 * $item->{indent};
+	my $bottom = '0';
+	if (defined $block->{stack}[ $state->{i} ] && $block->{stack}[ $state->{i} ]{type} eq 'par') {
+		$bottom = '1em';
+		$state->{i}++;
+	}
+	return $self->nl . "<div style=\"margin: 0 0 $bottom ${em}em;\">\n" . $self->string($item) . $self->nl . "</div>\n";
```

Comment:
If you really are trying to save individual characters being part of the HTML output, then remove the semi-colon after '${em}em' and the `\n` at the end, here.